### PR TITLE
Automatic link deteacking added

### PR DESCRIPTION
**Pull request by diminDDL**
Automatic link deteacking added in a separate cog, might still need a bit more testing and cleanup but that can be done in production :trollface:

### CODE REVIEW

File: thatkitebot/cogs/detrack.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        `+detrack` - toggles the setting server wide
+               
+        Only administrators and moderators can use this command.
+        """
+
+        if not await mods_can_change_settings(ctx):
+            return await ctx.respond("You don't have permission to change settings.")
+        
+        key = f"detrack:{ctx.guild.id}"
+        if await self.redis.exists(key) == 0:
```
Comment (ThatRedKite) [inline thatkitebot/cogs/detrack.py:132]: the `== 0 ` is redundant

File: thatkitebot/cogs/detrack.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            embed = discord.Embed()
+            embed.add_field(name="Auto link Detrack", value="Tracking links were detected, below are the detracked versions of the links.", inline=False)
+            for i in detrackedStrs:
+                embed.add_field(name="​", value=i, inline=False)
+            await message.reply(embed=embed)
+
+    @commands.check(mods_can_change_settings)
+    @bridge.bridge_command(name="detrack", aliases=["urlclean"],
+                           description="Remove known tracking urls and de-ampify links")
+    async def add_detrack_setting(self, ctx: commands.Context):
```
Comment (ThatRedKite) [inline thatkitebot/cogs/detrack.py:112]: that function name is really confusing

File: thatkitebot/cogs/detrack.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+        self.bot: thatkitebot.ThatKiteBot = bot
+        self.redis: aioredis.Redis = bot.redis
+        # this seems like a bad idea, maybe move it to __main__.py?
+        with open(os.path.join(bot.data_dir, "detrackparams.json"), "r") as f:
+            try:
+                self.detrack_data = json.load(f)
+            except json.decoder.JSONDecodeError:
+                print("detrackparams.json is not valid json. Please fix it.")
+
+    def process_dom_spec(self, filter, query):
```
Comment (ThatRedKite) [inline thatkitebot/cogs/detrack.py:26]: a really confusing function name. I have no idea what it is supposed to mean

**Comment by diminDDL** [comment]
time to pull :trollface:

**Comment by ThatRedKite** [comment]
i will regret this later :trollface:

**Pull request merged**